### DISCUSSION TITLE
Missing area in MagPIE 4.0 outputs for comparison

### DISCUSSION THREAD

**Discussion by acsoterroni** [magpiemodel/magpie#342] [General]
Hello everyone,

Very excited to be a new MagPIE user! I'm in the process of comparing the outputs from my local runs with the default runs. 

I did a quick check on the total land use available at global level by summing over all land use classes obtained from the R function `land` of `magpie4` library, and I found 3.57 Mha missing area in 2035. I did not get any missing area from my local runs (both 4.0 and 4.3.5 versions).

Am I doing something wrong? In any case, it is very small area (3.57/12805.13 = 0.03% of the total land).

Thank you!
Aline

`land("fulldata.gdx", file = NULL, level = "glo", types = NULL, subcategories = NULL, sum = FALSE, dir = ".", spamfiledirectory = "")`

**Comment by abhimishr** [comment]
Can you see if the following still returns some missing number in 2035
`dimSums(land(gdx),dim=c(1,3))`

I just checked from the SSP2 run results we submitted in the MAgPIE 4.0 paper and indeed 2035 seems to be off by 3.57Mha
```
dimSums(land(gdx),dim=c(1,3))
An object of class "magpie"
, , 1

      t
d1        y1995    y2000    y2005    y2010    y2015    y2020    y2025    y2030
  [1,] 12805.13 12805.13 12805.13 12805.13 12805.13 12805.13 12805.13 12805.13
      t
d1        y2035    y2040    y2045    y2050    y2055    y2060    y2070    y2080
  [1,] 12801.56 12805.13 12805.13 12805.13 12805.13 12805.13 12805.13 12805.13
      t
d1        y2090    y2100
  [1,] 12805.13 12805.13

```

cheers,
Abhi

**Reply by acsoterroni** [reply]
Hello,

Yes, it still returns a missing area in 2035 with `dimSums`.

Cheers,
Aline

**Comment by abhimishr** [comment]
Tagging @tscheypidi here. 

```
> round(dimSums(land(gdx),dim=3)[,"y2035",]- dimSums(land(gdx),dim=3)[,"y2030",],9)
An object of class "magpie"
, , 1

     t.t1
j      y2035.y2030
  CAZ  0.000000000
  CHA -3.568846151
  EUR  0.000000000
  IND  0.000000000
  JPN  0.000000000
  LAM -0.000000001
  MEA  0.000000000
  NEU  0.000000000
  OAS  0.000000000
  REF  0.000000000
  SSA  0.000000000
  USA  0.000000000
```
I notice a drop in area in China in 2035 (compared to other regions). But not a rounding issue, I guess? Something strange with NPIs?

**Reply by tscheypidi** [reply]
@acsoterroni welcome on board and interesting find. This is something we have not been aware of but I remember other cases in which we had similar behavior. Especially, as everything is back to normal in the following year I suspect that the optimizer made a mistake here, which happens sometimes (but not very often). Optimization was probably stopped before all variables had been set correctly.

In any case, do did everything correctly and there is indeed something off in the default run

**Reply by acsoterroni** [reply]
Thank you, Jan!
